functors_task_1/student/include/impl.h

```diff
@@ -1,13 +1,51 @@
 #pragma once
 #include "bar_serving.h"
 #include <functional>
+#include <iostream>
+#include <map>
+
+using namespace std;
```

Comment:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#sf6-use-using-namespace-directives-for-transition-for-foundation-libraries-such-as-std-or-within-a-local-scope-only

Please don't!